### ISSUE TITLE
Basilisk V3 Pro not appearing in Devices list

### ISSUE THREAD

**Issue by LadiesMan1309** [1kc/razer-macos#801]
<img width="1582" alt="Screen Shot 2023-06-10 at 10 31 07 PM" src="https://github.com/1kc/razer-macos/assets/80598601/c19d89b0-17b5-40bd-9fb2-77999e150447">
After downloading the application, my Basilisk V3 Pro device isn't appearing when searching for a device. I have tried connecting it via Bluetooth, a dongle, and plugging in a cable. All of this isn't working. Is there any way to fix this?

**Comment by nikhilsharma81181** [comment]
Any update on Razor Basilisk V3 Pro support??

**Comment by Thorazin** [comment]
Yep, just tried this tonight Saturday August 19th 2023. My Basilisk V3 Pro was not detected. Mac Studio Ultra, Ventura 13.5

**Comment by kldavis24** [comment]
Any chance we can get an update to if this support is being worked on, or a time frame?

**Comment by jakkornat** [comment]
Ready to review 🧑🏻‍💻: 
1. https://github.com/1kc/razer-macos/pull/818
2. https://github.com/1kc/librazermacos/pull/49

**Comment by NikhilPuulseDev** [comment]
@jakkornat Can we install this by any method??

**Comment by Frienducation7** [comment]
@jakkornat  Can we install this by any method??

**Comment by jakkornat** [comment]
@Frienducation7 until the PR isn't merged you can just clone the project directly from [my fork](https://github.com/jakkornat/razer-macos) - branch `basilisk_v3_pro` and follow the [guide for developers](https://github.com/jakkornat/razer-macos/tree/device/basilisk_v3_pro#developer-usage). 
It's worth mentioning my stack: 
* node `v16.20.2`
* yarn (newest from homebrew)
* Developer toolkit and clang - `Apple clang version 14.0.3`
* I used my own developer membership to sign the binary. You should use your own - I suppose you don't intend to publish, so those can be just a free developer membership credentials though.

**Comment by Frienducation7** [comment]
@jakkornat Thanks let me check this today

**Comment by maditamar** [comment]
Really appreciate all the work from everyone that's going into v3 pro integration. Just wanted to check in and see if any progress has been made towards a new .dmg for people who don't know how to assemble the package manually :) really wanna get my mouse workin!!!

**Comment by jakkornat** [comment]
I think that the only man who are able to merge my PR is @1kc but I can't catch him here by any chance 🤷🏻‍♂️.

**Comment by maditamar** [comment]
ahh ok :/ i dont know anything about how to merge so i guess ill just have to wait. is it a pretty complex process or is it something i could probably learn?

**Comment by Appzelof** [comment]
> @Frienducation7 until the PR isn't merged you can just clone the project directly from [my fork](https://github.com/jakkornat/razer-macos) - branch `basilisk_v3_pro` and follow the [guide for developers](https://github.com/jakkornat/razer-macos/tree/device/basilisk_v3_pro#developer-usage). It's worth mentioning my stack:
> 
> * node `v16.20.2`
> * yarn (newest from homebrew)
> * Developer toolkit and clang - `Apple clang version 14.0.3`
> * I used my own developer membership to sign the binary. You should use your own - I suppose you don't intend to publish, so those can be just a free developer membership credentials though.

I cloned your fork/project and ran `sh release.sh`. Basilisk v3 pro still won't appear in the device list.

I am using a 2023 MacBook ( M2 chip )

**Comment by jakkornat** [comment]
I haven't merged the local submodule there 😣. Please pull the changes with all submodules and try again. 

For the rest of you - please be aware that I'm working on a native Mac app dedicated for Basillist V3 Pro. I'll let you know when I will release it.

**Comment by jakkornat** [comment]
@Appzelof also, remember to first clear the device list and connect the mouse with a cable first.

**Comment by Appzelof** [comment]
@jakkornat Thank you! I got it to work by pulling all the submodules.

**Comment by jlandrum** [comment]
Just got a Basalisk V3 Pro; the Device ID when wireless is 0x00AC not 0x00AB, possibly a revision. Changing it, I was still unable to get it to work unfortunately; possibly due to other changes.

**Comment by jlandrum** [comment]
@jakkornat - after digging into it I figured out the issue - the submodule in your fork points to an older version of librazermacos which lacks support for Basilisk V3 Pro; moving that submodule to master fixed the issue (though for me, wireless still doesn't work)

**Comment by Pjmuscatella** [comment]
Ive been trying to get my Basilisk V3 Pro working on my MacBook M2. Just created a GitHub account to be able to follow updates, post, reply, etc....
I was able to get my BV3 Pro to show up in macOS device list only when wired, and after Sonoma update it's not recognized at all. 
It seems like there is a post or two in here with successful fixes to issues I’m experiencing, but I have no idea about any of this coding.
Is there anyone in here that has their Basilisk V3 Pro detected wirelessly, hopefully bluetooth but I'll take whatever, who can give me a breakdown of what steps I need to take from start to finish.
Maybe I'll get lucky and someone has it all laid out.
All I want to get is my lights and buttons configured to multiple profiles that are stored on the BV3Pro mouse, and connected to my MacBook via bluetooth so I can see the pretty flashing lights without dongles or wires sticking off my laptop.
A few days ago I would think this is such a simple request, but now I see how complicated it all is.
Its been very humbling!

**Comment by jlandrum** [comment]
@Pjmuscatella this software is generally just for the configuration side of things, it won't help you connect via Bluetooth.

Make sure the switch on the bottom is set to Bluetooth (towards the front) - if it isn't connected to anything it should start pairing mode (the Razer logo should pulse blue)

**Comment by Pjmuscatella** [comment]
What does 'Pull the submodules" mean?

**Comment by nikhilsharma81181** [comment]
Use "OpenRGB" app for RGB customization and "SteerMouse 5" for button customization both app are available for mac,
I hope it helps for non-programmer/normal-users

**Comment by Pjmuscatella** [comment]
Thanks Nikhilsharma.
Ive already downloaded OpenRGB as well.  OpenRGB recognizes the mouse, but mouse is unresponsive when connected.
But I appreciate the help.
It seems like a few people in some posts have resolved the issue with MacOS so I will continue to try and troubleshoot MacOS.
Any additional help with adjusting  the MacOS current download to work with Basilisk V3 pro will help

**Comment by Pjmuscatella** [comment]
@jlandrm Thanks for the reply,
I have no problem pairing bluetooth, and I would like to use MacOS to configure things. I would love for the BV3Pro to connect to my Mac anyway possible, but it just won't recognize the mouse.
Again thanks for the reply. I appreciate it.

**Comment by nikhilsharma81181** [comment]
@Pjmuscatella open OpenRGB app use your trackpad to configure settings and all then close the app and you can use it regularly without any problem, it works for me

**Comment by Pjmuscatella** [comment]
@nikhilsharma81181 I appreciate it. I actually have macOS working finally. Thanks for the advice!

**Comment by sergeyustyak** [comment]
I tried the version from the post below and it works with Basilisk V3 Pro
https://github.com/1kc/razer-macos/issues/778#issuecomment-1579648255

**Comment by christophbrejla** [comment]
would be great to have this merged to the main stream
